src/scripts/exclusion_term_expansion/exclusion_term_expansion.py

```diff
@@ -0,0 +1,299 @@
+"""Takes in the full ontology and the exclusions tsv and extracts a simple list of terms from it.
+Outputs two files:
+1. a simple list of terms (config/ONTO_NAME_term_exclusions.txt)
+2. a simple two column file with the term_id and the exclusion reason
+
+# Resources
+- GitHub issue: https://github.com/monarch-initiative/mondo-ingest/issues/22
+
+TODO's
+  - todo: Do we need to worry about edge case where exclusion table is empty? (e.g. we include such a table for all
+     ontologies, but there exists 1 where we don't exclude anything.
+  - todo: later: see below: #x1: regex results: would be good to filter out by IRI or prefix that is not from source
```

Comment:
`todo: later: see below: #x1: regex results: would be good to filter out by IRI or prefix that is not from source ontology, but how to determine the appropriate prefix just based on the 2 file params, but could have edge cases.`
Basically, it is possible that the REGEX matches might return terms that are in appear in one of our target ontologies, but are xrefs to another ontology. So I'm wondering if we might need to write some logic (and possibly pass an explicit param, e.g. the prefix to look for) to handle those edge cases.

Reply:
I hear you, but I don't know off the top of my head how to write a SPARQL query (or use one of our libraries, for that matter) to avoid this edge case. Maybe you have some idea?

Here's the SPARQL I have now for this query now:
```sparql
prefix owl:  <http://www.w3.org/2002/07/owl#>
prefix skos: <http://www.w3.org/2004/02/skos/core#>

select ?term_id ?term_label where {
  ?term_id a owl:Class;
    skos:prefLabel ?term_label .
  FILTER (REGEX(?term_label, "{{ regexp }}"))
  # FILTER(isIRI(?term_id) && (STRSTARTS(str(?term_id), "{{ iri }}")))  # Would be good to filter out xrefs from other ontologies. This could be done here or in Python.
}
```

The edge case I'm worried about would be caused by `where {?term_id a owl:Class; ...`

So actually, you are correct to be confused about the edge case catching xrefs. This wouldn't capture xrefs. But it still could capture instances where a class from a separate ontology is listed. I don't know why this would occur, but it still could theoretically happen, right?

Could avoid this by checking if the class is a descendant of the root node of the target ontology, but that would require this script to be more complex and we'd have to store the root node somewhere and pass it as a param.

We could also just not worry about this if we think it won't happen; up to you.

Reply:
Oh, the expansion  process should _only_ use rdfs:subClassOf and owl:equivalentClass, no other relation (in particular not xref). Is this what you are asking?

Reply:
I was just asking about how to filter out so only classes that belong to that ontology are included in the results. But you actually recommended a good solution for this earlier; you had me use the `signature-%` files. So I believe this is resolved.